include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1818,8 +1650,13 @@ __parallel_sort_impl(_ExecutionPolicy&& __exec, _Range&& __rng, _Merge __merge,
     using _CopyBackKernel =
         oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<__sort_copy_back_kernel<_CustomName>>;
 
-    return __parallel_sort_submitter<_LeafSortKernel, _GlobalSortKernel, _CopyBackKernel>()(
-        ::std::forward<_ExecutionPolicy>(__exec), ::std::forward<_Range>(__rng), __merge, __comp);
+    const auto __n = __rng.size();
+    if(__n <= std::numeric_limits<::std::uint32_t>::max())
+        return __parallel_sort_submitter<::std::uint32_t, _LeafSortKernel, _GlobalSortKernel, _CopyBackKernel>()(
+            ::std::forward<_ExecutionPolicy>(__exec), ::std::forward<_Range>(__rng), __comp);
+    else
+        return __parallel_sort_submitter<::std::uint64_t, _LeafSortKernel, _GlobalSortKernel, _CopyBackKernel>()(
+            ::std::forward<_ExecutionPolicy>(__exec), ::std::forward<_Range>(__rng), __comp);
```

Comment:
There is a problem with kernel redifinition.
 This fixes it: [kernel_names.patch](https://github.com/oneapi-src/oneDPL/files/12463987/kernel_names.patch)

Reply:
Do you know if this is the cause of the failing CI?

Reply:
No, I guess it is another issue

Reply:
fixed.